### ISSUE TITLE
decoration::blur::enabled = 1 Makes Max FPS Half of Monitor Refresh Rate

### ISSUE THREAD

**Issue by TZProgrammer** [hyprwm/Hyprland#4438]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit b240704bee1d04eacc27654ea875cf6f15033c7c dirty (renderer: allow rendering multiple fullscreen windows in third fs pass).
Date: Tue Jan 9 14:42:07 2024
Tag: v0.34.0-37-gb240704b
```

</details>

### Bug or Regression?

Bug

### Description

Upon setting decoration::blur::enabled 1, my FPS immediately goes from the maximum (refresh rate of the monitor) to exactly half of that. In my case it went from 144 to 72.

I highly doubt that this is just a "blue is intensive, of course FPS is lower" type of scenario. If that were the case, it would not make sense for the FPS to be exactly half of the max refresh rate, or for it to  be so stable around that point.

Oh, also, nvidia with anti_flicker off. Enabling that option just makes my FPS non-existent, so I have not checked with it on

### How to reproduce

Start glxgears. toggle decoration::blur::enabled 1, and wait for FPS to stabilize. Toggle it back to 0, wait for the FPS to stabilize. On my system, the end FPS with blur was exactly half of the refresh rate.

### Crash reports, logs, images, videos

![image](https://github.com/hyprwm/Hyprland/assets/22439185/0afde42b-44f2-49d4-afb4-b25c7bbf0d61)

In the beginning, blur was enabled, FPS = 72. When it was disabled, FPS rose to the refresh rate. Later, I enabled it again, FPS halved.

**Comment by vaxerski** [comment]
> If that were the case, it would not make sense for the FPS to be exactly half of the max refresh rate, or for it to be so stable around that point.

kinda would. Why don't you post `hyprctl systeminfo` as requested?

**Comment by TZProgrammer** [comment]
"hyprctl systeminfo" is simply not working for me, unfortunately, prints the "usage: hyprctl" info.

I would like to understand why it makes sense for blur to halve the FPS to exactly half of the refresh rate. From my perspective, if it was a bit more than half, or a bit less, I would agree. But for it to be *exactly* half, like, 72.00, and hold that framerate for a minute straight with barely any deviations, seems like it would be far too unlikely a coincidence.

**Comment by vaxerski** [comment]
partial upgrades moment?

**Comment by vaxerski** [comment]
> I would like to understand why it makes sense for blur to halve the FPS to exactly half of the refresh rate. From my perspective, if it was a bit more than half, or a bit less, I would agree. But for it to be exactly half, like, 72.00, and hold that framerate for a minute straight with barely any deviations, seems like it would be far too unlikely a coincidence.

well learn about how graphics stacks in vsync applications work, especially wayland. That should help understand why this happens.

**Comment by TZProgrammer** [comment]
Yeah, I think you might be right.

**Comment by kolan4ick** [comment]
Did you manage to solve this problem? I have the same situation and I saw that the problem is due to the fact that the hyprland session is running on the iGPU. Yes, as I have two iGPU graphics cards from Intel and nvidia GPU on my laptop. I was not able to solve this problem and make hyprland run on nvidia. The nvidia-smi command shows that Hyprland is using 1mib. The only solution I found was to increase the frequency of the integrated graphics card from 300-350 by default to 900. This could be a solution, but it always makes me use the fans, which is annoying.
